### PR TITLE
loader: Do not enumerate extensions from disabled implicit layers

### PR DESCRIPTION
**Pull request by jfroy**
This patch adds a check to vkEnumerateInstanceExtensionProperties to
omit extensions provided by disabled implicit layers. Per section 31.2
(1.0.51), "When pLayerName parameter is NULL, only extensions provided
by the Vulkan implementation or by implicitly enabled layers are
returned." Arguably, implicit layers that have been disabled are not
"implicitly enabled".

### CODE REVIEW

File: loader/trampoline.c [hunk 1]
```diff
@@ -143,9 +143,10 @@ LOADER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkEnumerateInstanceExtensionPropert
         }
         loader_scanned_icd_clear(NULL, &icd_tramp_list);
 
-        // Append implicit layers.
+        // Append enabled implicit layers.
         loader_implicit_layer_scan(NULL, &instance_layers);
         for (uint32_t i = 0; i < instance_layers.count; i++) {
+            if (!loader_is_implicit_layer_enabled(NULL, &instance_layers.list[i])) continue;
```
Comment (MarkY-LunarG) [inline loader/trampoline.c]: Nit: Our coding standard requires braces.  Can you put the continue on a new line and open brace on the end of this and close brace after?

Comment (jfroy) [inline loader/trampoline.c]: Sure can.

**Comment by mark-lunarg** [comment]
@jfroy, @MarkY-LunarG, these changes look fine to me as well, but we'll need a clean rebase (without the merge) in order to take this automatically.  Or, Mark can do this manually -- it's up to you guys.

**Comment by jfroy** [comment]
I can certainly rebase.

**Pull request merged**